kernel/Kconfig

```diff
@@ -922,6 +922,7 @@ config TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
 config THREAD_LOCAL_STORAGE
 	bool "Thread Local Storage (TLS)"
 	depends on ARCH_HAS_THREAD_LOCAL_STORAGE && TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
+	default y if ARM || RISCV
```

Comment:
Is it better that let arm arch or riscv arch Kconfig select the `TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE`, instead of using `default y if ARM || RISCV`

Reply:
This patch is marked DNM because it's purely for testing these fixes in CI. I suspect enabling TLS by default for any system is a much longer discussion.

Reply:
On the suggestion of @andyross I changed this from enabling TLS on all arm and riscv architectures to enabling it only for the qemu_riscv32 boards -- this should provide a reasonable level of CI test coverage without impacting any existing projects using real hardware.